lib/bot-service.js

```diff
@@ -26,7 +26,16 @@ botService.get = async (resource) => {
             });
             response.on('end', () => {
                 const data = decodeToken(body);
-                data === undefined ? reject() : resolve(data);
+                var responseBody = {};
+                if (data instanceof Error)  {
```

Comment:
does this ever happen?

Reply:
Yes, this happens in either one of these 2 situations - Missing key `bot` in response JSON or Unable to decode response from BoT server.

Reply:
I think the decodeToken function should throw that error then, and not here. Now we have it in two places.

Reply:
Changed in decodeToken to throw error and removed checking for error within get/post